wordpress-mobile/WordPress-Android
# Track Blogging Reminders `notification_tapped` event

This PR adds event tracking for clicks on the Blogging Reminders notification.

It adds a `blogging_reminders` [notification type](https://github.com/wordpress-mobile/WordPress-Android/blob/8c47ccdb0ab371a8bb0d63f5448595d4040c6fd2/WordPress/src/main/java/org/wordpress/android/push/NotificationType.kt) and passes that information as a parameter to the reminder notification intent. When the `PostsListActivity` is opened from the notification, it checks for that parameter and tracks the event.

### To test

To test if the notification is shown, I recommend updating the following line with a shorter delay (30 seconds, for example):

https://github.com/wordpress-mobile/WordPress-Android/blob/a53343577d029ea731fab9ca814bd0ddc13547b2/WordPress/src/main/java/org/wordpress/android/workers/reminder/ReminderScheduler.kt#L31

1. Clear app data and login.
1. On the My Site screen, tap "Settings".
1. On the Site Settings screen, tap "Blogging Reminders".
1. Go through the Blogging Reminders flow and select a few days.
1. Notice the Blogging Reminders notification after the initial delay.
1. After the notification is shown, click on it.
1. Notice the Posts List screen.
1. On `logcat`, make sure you see a message like:

```
 🔵 Tracked: notification_tapped, Properties: {"notification_type":"blogging_reminders"}
```

## Regression Notes
1. Potential unintended areas of impact
None that I can think of.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
